File: backend/app/Models/HomeSection.php
<?php

declare(strict_types=1);

namespace App\Models;

/**
 * Home-screen section (DATABASE_DESIGN.md §L.8).
 *
 * A server-driven, ordered home block enabling a fully dynamic home layout
 * without app releases. Unknown `section_type`s are ignored gracefully by the
 * client (forward-compatible).
 */
final class HomeSection extends BaseModel
{
    protected string $table = 'home_sections';

    /** @var array<int, string> */
    protected array $fillable = [
        'section_type',
        'title',
        'config',
        'sort_order',
        'target_audience',
        'is_active',
        'starts_at',
        'ends_at',
    ];

    /** @var array<string, string> */
    protected array $casts = [
        'id'              => 'int',
        'sort_order'      => 'int',
        'is_active'       => 'bool',
        'config'          => 'json',
        'target_audience' => 'json',
    ];

    public function sectionType(): string
    {
        $type = $this->get('section_type');

        return is_string($type) ? $type : '';
    }
}

File: mobile/lib/features/home/models/home_section.dart
import 'package:flutter/foundation.dart';

/// A server-driven home layout section (`GET /v1/home/layout`).
///
/// `type` is kept as a raw string so unknown/newer section types are ignored
/// gracefully by the renderer (forward-compatible).
@immutable
class HomeSection {
  const HomeSection({
    required this.type,
    required this.sortOrder,
    this.title,
    this.config = const {},
  });

  final String type;
  final int sortOrder;
  final String? title;
  final Map<String, dynamic> config;

  factory HomeSection.fromJson(Map<String, dynamic> json) {
    final config = json['config'];
    return HomeSection(
      type: json['type'] as String? ?? '',
      sortOrder: (json['sort_order'] as num?)?.toInt() ?? 0,
      title: json['title'] as String?,
      config: config is Map<String, dynamic> ? config : const {},
    );
  }

  /// Reads a string list from the section config (e.g. quick-action keys).
  List<String> stringList(String key) {
    final value = config[key];
    if (value is List) {
      return value.whereType<Object>().map((e) => e.toString()).toList();
    }
    return const [];
  }
}
